oauth2/oauth2_auth_code_test.go

```diff
@@ -392,6 +392,41 @@ func TestAuthCodeWithDefaultStrategy(t *testing.T) {
 								_, err := oauthConfig.TokenSource(oauth2.NoContext, token).Token()
 								require.Error(t, err)
 							},
+						}, {
+							d:       "Checks if request fails when subject doesn't match",
+							authURL: oauthConfig.AuthCodeURL("some-hardcoded-state", oauth2.SetAuthURLParam("audience", "https://not-ory-api/")),
+							cj:      newCookieJar(),
+							lph: func(t *testing.T) func(w http.ResponseWriter, r *http.Request) {
+								return func(w http.ResponseWriter, r *http.Request) {
+									_, res, err := apiClient.GetLoginRequest(r.URL.Query().Get("login_challenge"))
+									require.NoError(t, err)
+									require.EqualValues(t, http.StatusOK, res.StatusCode)
```

Comment:
... and add a `assert.False(t, challenge.Skip)`. This will make sure that the test is actually failing because the subject is empty, not because the subjects don't match.